src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -1415,15 +1415,192 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
+                // the output to be in the same xmm register
+                // Hence we clone the first operand
+                GenTree* op2Clone = fgMakeMultiUse(&op1);
+                // GenTree* op2Clone;
+                // op1 = impCloneExpr(op1, &op2Clone, CHECK_SPILL_ALL,
+                //                     nullptr DEBUGARG("Cloning double for Dbl2Ulng conversion"));
+                
+                //run vfixupimmsd base on table and no flags reporting
+                GenTree* saturate_val = gtNewSimdHWIntrinsicNode(simdType, op1, op2Clone, tbl, gtNewIconNode(0),
+                                                            NI_AVX512F_Fixup, simdBaseJitType, simdSize);
+
+                GenTree* max_val = gtNewSimdCreateBroadcastNode(simdType, gtNewDconNodeD(static_cast<double>(INT64_MAX)), simdBaseJitType, simdSize);
+                GenTree* max_valDup = gtNewSimdCreateBroadcastNode(simdType, gtNewIconNode(INT64_MAX, TYP_LONG), CORINFO_TYPE_LONG, simdSize);
+                //we will be using the input value twice
+                GenTree* saturate_valDup = fgMakeMultiUse(&saturate_val);
+
+                //usage 1 --> compare with max value of integer
+                saturate_val = gtNewSimdCmpOpNode(GT_GE, simdType, saturate_val, max_val, simdBaseJitType, simdSize);
+                //cast it
+
+                intrinsic = (simdSize == 16) ? NI_AVX512DQ_VL_ConvertToVector128Int64WithTruncation
+                                             : (simdSize == 32) ? NI_AVX512DQ_VL_ConvertToVector256Int64WithTruncation
+                                                                : NI_AVX512DQ_ConvertToVector512Int64WithTruncation;
+
+                retNode = gtNewSimdHWIntrinsicNode(retType, saturate_valDup, intrinsic, simdBaseJitType, simdSize);
+
+                //usage 2 --> use thecompared mask with input value and max value to blend
+                retNode = gtNewSimdCndSelNode(simdType, saturate_val, max_valDup, retNode, CORINFO_TYPE_LONG, simdSize);
+            }
+#endif // TARGET_AMD64
+            break;
+        }
+
         case NI_Vector128_ConvertToUInt32:
         case NI_Vector256_ConvertToUInt32:
+        case NI_Vector512_ConvertToUInt32:
+        {
+            assert(sig->numArgs == 1);
+#ifdef TARGET_AMD64
+            assert(varTypeIsFloating(simdBaseType));
+            if (IsBaselineVector512IsaSupportedOpportunistically())
+            {
+                op1     = impSIMDPopStack();
+
+                var_types simdType = getSIMDTypeForSize(simdSize);
+                // Generate the control table for VFIXUPIMMSD
+                // The behavior we want is to saturate negative values to 0.
+                GenTreeVecCon* tbl = gtNewVconNode(simdType);
+
+                // QNAN: 0b0000:
+                // SNAN: 0b0000
+                // ZERO: 0b0000:
+                // +ONE: 0b0000
+                // -INF: 0b0000
```

Comment:
Does `-VAL` include `-INF` or is it better to cover it explicitly here?

Reply:
Even without handling the ```-inf``` and ```+inf``` they work as intended. I dont think it will need separate handling since the current implementation handles those values and saturates them in the correct manner. But for safety and clear code we can add them to ```vfixupimmsd/ss``` control table. Let me know what you think.

Reply:
I think this was handled in my latest changes as you suggested. There was not test running on -inf so all test passed when -inf was not handled. The latest code will have all the things handled.